app/setup/index.js

```diff
@@ -0,0 +1,259 @@
+            if (res.status !== 200) {
+                this.setState({status: 'failure', errorMessage: json.error.message});
+            }
+            this.setState({loggedIn: json.hasAuth});
+        }));
+    }
+
+    setTimeoutAuthDone () {
+        const {username} = this.state;
+        const checkAuth = setInterval(() => {
+            // TODO: This is not very clear for a message. Show them a link to the
+            // maybe in case they closed it or want to try again?
+            this.setState({
+                errorMessage: `We\'re waiting for authorization of [${username}].`});
+            this.verifyAuthDone();
+            if (this.state.loggedIn) {
+                // TODO: Is there a better way? On premise has the `external-data-connector`
+                // route so cant use window.location.origin.
+                clearInterval(checkAuth);
+                const appRoute = baseUrl().replace('/setup', '/');
+                window.location.assign(appRoute);
+            }
+        }, 1000);
+    }
+
+    authenticateUser () {
+        if (!this.state.username) {
+            this.setState({
+                status: 'failure',
+                errorMessage: 'Plotly username can not be empty.'
+            });
+            return;
+        }
+        if (!this.state.domain && this.state.serverType === ONPREM) {
+            this.setState({
+                status: 'failure',
+                errorMessage: 'Plotly On Premise domain can not be empty.'
+            });
+            return;
+        }
+        this.setState({errorMessage: ''});
+        this.setTimeoutAuthDone();
+        if (isWebBrowser) {
+            window.open(this.buildOauthUrl(), '_blank');
+        } else {
+            dynamicRequireElectron().shell.openExternal(this.buildOauthUrl());
+        }
+    }
+
+    updateStateWithEvent(e) {
+        this.setState({[e.target.name]: e.target.value});
+    }
+
+    render() {
+        const {errorMessage, status} = this.state;
+        const submitButton = (value) => {
+            return (
+                <button
+                    onClick={(e) => this.chooseServerType(e.target.value)}
+                    className="btn btn-large btn-primary"
+                    value={value}
+                    style={{margin: '10px'}}
+                >
+                    {SERVER_TYPES[value]}
+                </button>
+            );
+        };
+
+        const loginButton = () => {
+            return (
+                <button
+                    className="btn btn-large btn-primary"
+                    style={{display: 'block', margin: 'auto'}}
+                    onClick={() => this.authenticateUser()}
+                >{'Login'}</button>
+            );
+        };
+
+        const chooseServerType = () => {
+            return (
+                <div className="control-group">
+                <h3 className="block-center-heading">
+                    Choose Your Plotly Use Case
```

Comment:
It's not really a `Use Case`. Also, lots of plotly cloud users don't even know that they are "plotly cloud" users and don't know what "On Premise" is.

Maybe we can make this clearer by having `Plotly Cloud` be the default in some way and then allowing `Plotly On Premise` users to click something that allows them to change the default URL

```
<label>Your Plotly Username</label>
<input/>

<label>Log in to</label>
<radio selected>Plotly Cloud (https://plot.ly)</radio>
<radio>Plotly On-Premise</radio>

<button>Log In</button>
```

The `Plotly Cloud` radio item could be selected by default. If the user selects the Plotly On-Premise radio button, a text input would appear that would allow them enter their plotly on-premise domain.